pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -105,20 +103,29 @@ func (th *TopologyHint) IsEqual(topologyHint TopologyHint) bool {
 	return false
 }
 
-// String converts TopologyHint to string
-func (th *TopologyHint) String() string {
-	nodeAffinity := "nil"
-	if th.NUMANodeAffinity != nil {
-		nodeAffinity = th.NUMANodeAffinity.String()
+// LessThan checks if TopologyHint `a` is less than TopologyHint `b`
+// this means that either `a` is a preferred hint and `b` is not
+// or `a` NUMANodeAffinity attribute is narrower than `b` NUMANodeAffinity attribute.
+func (th *TopologyHint) LessThan(other TopologyHint) bool {
+	if th.Preferred != other.Preferred {
+		return th.Preferred == true
 	}
-	return fmt.Sprintf("(Preferred=%t, NUMANodeAffinity=%s)", th.Preferred, nodeAffinity)
+	return th.NUMANodeAffinity.IsNarrowerThan(b.NUMANodeAffinity)
```

Comment:
'b.NUMANodeAffinity' should be 'other.NUMANodeAffinity' ?

Reply:
Yes! will sort it out